config/sql/config.go

```diff
@@ -93,4 +83,19 @@ func Configure(p *config.Provider) {
 		)
 		r.UseAsync = true
 	})
+	p.AddResourceConfigurator("azurerm_mssql_server_transparent_data_encryption", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_id": config.Reference{
+				Type:      "MSSQLServer",
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+			"key_vault_key_id": config.Reference{
+				Type:      rconfig.APISPackagePath + "/keyvault/v1alpha2.Key",
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+		}
+		r.ExternalName = config.IdentifierFromProvider
+		r.UseAsync = true
```

Comment:
Do we have an actual timing for the provisioning of these resources? Using the sync mode is more performant but using the async mode is more safe. Personally I prefer to be on the safe side, just asking to make this explicit.

Reply:
@ulucinar what is a reliable way to measure performance? 
Is `time kubectl apply -f $resource.yaml` a reliable way in case of Sync mode?

Reply:
Timing `kubectl` would not be suitable as it just measures the time it takes to store the CR in etcd. What we are interested in this context is the time it takes to actually provision the remote (infrastructure) resource. If we are using the sync mode and if it takes longer than the controller's reconcile context timeout for the resource to be provisioned (i.e., for the `terraform apply` to report either success or failure), then we interrupt it. So the time we need to measure is the time to actually provision the resource (the time it takes a `terraform apply` to provision it). 
I think we can continue with the async mode if you'd like to, just mentioned it for future discussions or optimizations.

Reply:
got it, I will try to measure async vs sync scenarios

Reply:
Thanks @ytsarev. As I mentioned, I'm fine with the async mode, it is more of an informative comment.

Reply:
@ulucinar async vs sync tests:

Async - 26 seconds:
```
k apply -f ./mssqlservertransparentdataencryption.yaml;date
Wed Mar 30 15:07:45 CEST 2022

 k get mssqlservertransparentdataencryption.sql.azure.jet.crossplane.io/transparent-encryption-example -o yaml
  - lastTransitionTime: "2022-03-30T13:08:11Z"
    reason: Available
    status: "True"
    type: Ready
```

Sync - 23 seconds :
```
k apply -f ./mssqlservertransparentdataencryption.yaml;date
Wed Mar 30 15:14:10 CEST 2022

k get mssqlservertransparentdataencryption.sql.azure.jet.crossplane.io/transparent-encryption-example -o yaml
....
    Last Transition Time:  2022-03-30T13:14:33Z
    Reason:                Available
    Status:                True
    Type:                  Ready
```

Where do we set the reconciliation context timeout?

Reply:
Hi @ytsarev,
I think then we can use the synchronous mode for this resource then. The context deadline is determined in the generated `internal/controller/sql/mssqlservertransparentdataencryption/zz_controller.go` file [here](https://github.com/crossplane-contrib/provider-jet-azure/blob/cc519526841562594e8cbe1c77b895aac1e570af/internal/controller/sql/mssqlservertransparentdataencryption/zz_controller.go#L47). So it's not overridable per resource-type.

Reply:
cool, looks safe for then Sync mode indeed. I will amend the PR